cpp/src/arrow/csv/converter.h

```diff
@@ -57,7 +57,9 @@ class ARROW_EXPORT Converter {
 
   virtual Status Initialize() = 0;
 
-  const ConvertOptions options_;
+  // CAUTION: ConvertOptions can grow large (if it customizes hundreds or
+  // thousands of columns), so avoid copying it in each Converter.
+  const ConvertOptions& options_;
```

Comment:
I'd say keep this one a value, but let the ColumnBuilder's be reference, since they're component inside the Converter. This would amount to a single copy (which can be moved' by the caller), or none if you move in the constructor.

Reply:
I don't think this would fix the issue, since there is one converter per column.

Reply:
urgh, I was under the impression that there was one converted with many ColumnBuilders :(